File: services/api/app/schemas/usage.py
from datetime import date

from pydantic import BaseModel


class UsagePeriodResponse(BaseModel):
    """Activity in one calendar month plus the limits in force (0 = unlimited)."""

    period_start: date
    analyses_count: int
    image_count: int
    text_count: int
    reports_count: int
    provider_calls_count: int
    provider_cost: float
    storage_bytes_period: int
    # Bytes currently held for the user (live, unpurged originals, reports, artifacts).
    current_storage_bytes: int
    analysis_limit: int
    storage_limit_bytes: int
    provider_cost_limit: float
    analyses_remaining: int | None
    storage_remaining_bytes: int | None
    provider_budget_remaining: float | None


class UsageHistoryResponse(BaseModel):
    periods: list[UsagePeriodResponse]

File: packages/shared-types/src/usage.ts
/** Mirrors `app/schemas/usage.py`. Limits of 0 mean unlimited. */

export interface UsagePeriodResponse {
  period_start: string;
  analyses_count: number;
  image_count: number;
  text_count: number;
  reports_count: number;
  provider_calls_count: number;
  provider_cost: number;
  storage_bytes_period: number;
  /** Bytes currently held for the user (live, unpurged originals, reports, artifacts). */
  current_storage_bytes: number;
  analysis_limit: number;
  storage_limit_bytes: number;
  provider_cost_limit: number;
  analyses_remaining: number | null;
  storage_remaining_bytes: number | null;
  provider_budget_remaining: number | null;
}

export interface UsageHistoryResponse {
  periods: UsagePeriodResponse[];
}
